raiden/raiden_service.py

```diff
@@ -619,6 +633,141 @@ def settle(self, token_address, partner_address):
         return netting_channel
 
 
+class ConnectionManager(object):
+    """
+    The ConnectionManager provides a high level abstraction for connecting to a
+    Token network.
+    """
+    I_AM_HERE_ADDR = '2' * 40
+
+    def __init__(
+        self,
+        raiden,
+        token_address,
+        initial_channel_target=3,  # number of channels to open immediately
+        joinable_funds_target=.4,  # amount of funds not initially assigned
+    ):
+        self.lock = Semaphore()
+        self.funds = 0
+        self.raiden = raiden
+        self.token_address = safe_address_decode(token_address)
+        self.initial_channel_target = initial_channel_target
+        self.joinable_funds_target = joinable_funds_target
+        # force state update
+        self.raiden.event_handler.poll_all_event_listeners()
+        try:
+            self.manager = self.raiden.get_manager_by_token_address(self.token_address)
+        except UnknownTokenAddress:
+            log.debug("token not yet registered", token=pex(self.token_address))
+            self.raiden.chain.default_registry.add_token(self.token_address)
+            self.raiden.event_handler.poll_all_event_listeners()
+            self.manager = self.raiden.get_manager_by_token_address(self.token_address)
+            # make ourselves visible
+            self.raiden.api.open(
```

Comment:
> Then there should be no UnknownTokenException.
I thought the exception handling was for some sort of racing condition, can we just check if the address is in the `managers_by_token_address` dictionary?